### DISCUSSION TITLE
[7.x] isPlaying() returns false just after PlayPause()

### DISCUSSION THREAD

**Discussion by haivala** [katspaugh/wavesurfer.js#3048] [Q&A]
## Bug description
Am I only one with this problem?

## Environment
 - Browser: Vivaldi

## Minimal code snippet
```
this.ws.on("ready", () => {
    this.ws.playPause();
    console.log(this.ws.isPlaying()); // returns false
});
```
## Expected result
isPlaying() returns true

## Obtained result
isPlaying() returns false

**Comment by katspaugh** [comment]
This is the expected behavior.

> The [HTMLMediaElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement) play() method attempts to begin playback of the media. It returns a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) which is resolved when playback has been successfully started.

See https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/play

You can fix this in your code by waiting for play to resolve:

```js
this.ws.on("ready", async () => {
    await this.ws.playPause();
    console.log(this.ws.isPlaying());
});
```

**Reply by haivala** [reply]
This did not change the output.

I resulted to check if the player is playing first and the control it. So that I did not have to use the `playPause()`

**Comment by doesdev** [comment]
It's worth noting that calling `isPlaying` right after the `play` event first fires also returns false. That would seem like a bug, no? Userland code should definitely await the `play` Promise but the event should only fire when playback has begun I would think.

**Reply by doesdev** [reply]
Well, there's more to it than that actually. You're proxying the `play` event from the MediaElerment API which fires when the `pause` property changes to true. I thought perhaps the `playing` event would work instead but `isPlaying` is also false there on first playback.

Seems `isPlaying` isn't in agreement with the `paused` state. For my usage I'll switch to checking the pause state on the media element though it would be ideal to stick with properties intended to be public.

```js
ws.media.addEventListener('play', () => console.log({ event: 'play', isPlaying: ws.isPlaying(), paused: ws.media.paused }))
ws.media.addEventListener('playing', () => console.log({ event: 'playing', isPlaying: ws.isPlaying(), paused: ws.media.paused }))
```

**Reply by katspaugh** [reply]
Thanks for the analysis!
Perhaps we shouldn’t be checking that currentTime is not zero (https://github.com/katspaugh/wavesurfer.js/blob/main/src/player.ts#L98). That’s the only difference from media.paused.
